Jetpack Manage: Implement pricing summary in the review licenses modal

yashwin opened the pull request:
Resolves https://github.com/Automattic/jetpack-genesis/issues/98

## Proposed Changes

This PR displays the pricing summary in the review licenses modal.

## Testing Instructions

**Prerequisites**

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

**Instructions**

1. Visit the Jetpack Cloud link > Replace `dashboard` in the URL with `/partner-portal/issue-license?flags=jetpack/bundle-licensing`.
2. Select a few licenses and click the `Review X licenses` button > Verify that the total price is shown along with the `Issue X license` button on the right side of the modal. Please note the CTA & Learn more link doesn't do anything. It will be updated in upcoming PRs. The pricing breakdown will the done in another PR. The discounted price needs to be updated based on the selected bundle(TODO)

Large screen:

<img width="1154" alt="Screenshot 2023-11-24 at 11 11 06 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/2cb5314b-d138-45a4-88e1-a527ea01c213">

Tablet view:

<img width="591" alt="Screenshot 2023-11-24 at 11 24 21 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/8a52a3d0-90a9-480a-86ee-82276f5c280a">

Mobile view:

<img width="395" alt="Screenshot 2023-11-24 at 11 12 01 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/d7499fbf-74d8-4d79-8582-31f9a57954ea">




## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

yashwin committed:
Implement pricing summary for review licenses modal

jkguidaven commented on client/jetpack-cloud/sections/partner-portal/issue-license-v2/lib/pricing.ts:
```diff
@@ -0,0 +1,74 @@
+		if ( monthlyProduct ) {
+			const actualCost = isDailyPricing ? monthlyProduct.cost / 365 : monthlyProduct.cost;
+			const discountedCost = actualCost - productCost;
+			discountInfo.discountPercentage = ! productCost
```
[non-blocker] We can convert this to a positive condition.

```
discountInfo.discountPercentage = productCost
   ? Math.round( ( discountedCost / actualCost ) * 100 ) : 100;
```

jkguidaven commented on client/jetpack-cloud/sections/partner-portal/issue-license-v2/review-licenses/pricing-summary.tsx:
```diff
@@ -0,0 +1,65 @@
+	return (
+		<>
+			<div className="review-licenses__pricing">
+				<span className="review-licenses__price">
```
[non-blocker]I suggest to name this `review-licenses__pricing-discounted` and the other `review-licenses__pricing-original` to match the corresponding content.

jkguidaven reviewed (approved):
I confirmed that the price and CTA are properly rendered across different screen sizes. I left a few non-blocking comments, but otherwise everything is good. Nice work! 👍

yashwin committed:
Address PR comments

yashwin commented:
Thanks for the review & feedback. I have addressed your comments: https://github.com/Automattic/wp-calypso/pull/84496/commits/e298ae38641518c7250bf2e4f780c4f1d8767bf8. Could you please take another look? TIA 🙂

jkguidaven reviewed (approved):
Thanks for addressing my comments. Retested and works as expected. The new changes LGTM 👍

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9765006

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9765006).
